From f3d84027dc1ade535d2bd38f43df725fe7797838 Mon Sep 17 00:00:00 2001
From: html6405 <peter.schelchshorn@mhs-solutions.at>
Date: Mon, 25 Dec 2023 20:49:17 +0100
Subject: [PATCH 8/8] Revert "Rewrite renameat()."

We still want to support kernels without renameat2

This reverts commit 6660d6c00b2253a941aaa3fcd01ead885327b3b5.
---
 libc/SYSCALLS.TXT      | 1 +
 libc/bionic/rename.cpp | 6 +-----
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/libc/SYSCALLS.TXT b/libc/SYSCALLS.TXT
index 3622ca4f4..aa4eb4ed4 100644
--- a/libc/SYSCALLS.TXT
+++ b/libc/SYSCALLS.TXT
@@ -168,6 +168,7 @@ linkat(int, const char*, int, const char*, int)  all
 mkdirat(int, const char*, mode_t)  all
 mknodat(int, const char*, mode_t, dev_t)  all
 readlinkat(int, const char*, char*, size_t)  all
+renameat(int, const char*, int, const char*)  all
 renameat2(int, const char*, int, const char*, unsigned)  all
 symlinkat(const char*, int, const char*)  all
 unlinkat(int, const char*, int)   all
diff --git a/libc/bionic/rename.cpp b/libc/bionic/rename.cpp
index 89786f7a0..82955596b 100644
--- a/libc/bionic/rename.cpp
+++ b/libc/bionic/rename.cpp
@@ -30,9 +30,5 @@
 #include <stdio.h>
 
 int rename(const char* old_path, const char* new_path) {
-  return renameat2(AT_FDCWD, old_path, AT_FDCWD, new_path, 0);
-}
-
-int renameat(int old_dir_fd, const char* old_path, int new_dir_fd, const char* new_path) {
-  return renameat2(old_dir_fd, old_path, new_dir_fd, new_path, 0);
+  return renameat(AT_FDCWD, old_path, AT_FDCWD, new_path);
 }
-- 
2.51.0

